File: src/main/java/com/trektrip/service/CountryService.java
package com.trektrip.service;

import com.trektrip.model.Country;

import java.util.List;
import java.util.Optional;

public interface CountryService {
    List<Country> getAllCountries();
    Optional<Country> getCountryById(Long id);
    Country createCountry(Country country);
    Country updateCountry(Country country, Long id);
    void deleteCountry(Long id);
}

File: frontend/src/Services/countryService/countryService.js
import axiosInstance from "../../axios/axiosInstance";

const API_URL = '/country'; 

const getAllCountries = () => {
    return axiosInstance.get(`${API_URL}/all`);
};

const getCountryById = (id) => {
    return axiosInstance.get(`${API_URL}/${id}`);
};

const createCountry = (country) => {
    return axiosInstance.post(API_URL, country);
};

const updateCountry = (id, country) => {
    return axiosInstance.put(`${API_URL}/${id}`, country);
};

const deleteCountry = (id) => {
    return axiosInstance.delete(`${API_URL}/${id}`);
};

export default {
    getAllCountries,
    getCountryById,
    createCountry,
    updateCountry,
    deleteCountry,
};
